src/color.rs

```diff
@@ -126,6 +126,145 @@ pub struct Color {
+    /// Values that are outside the standard RGB or alpha range will result in a panic
+    /// 
+    /// ```should_panic
+    /// use turtle::Color;
+    /// // This will not work
+    /// let color = Color::rgba(255.0, 255.0, 255.0, 1.1);
+    /// ```
+    /// [`Color`]: ./index.html
+    pub fn rgba(red: f64, green: f64, blue: f64, alpha: f64) -> Self {
+        let color = Color { red, green, blue, alpha };
+        assert!(color.is_valid(), "Invalid color: {:?}. See the Color struct documentation for more information.", color);
+        color
+    }
+
+    /// Create a new `Color` from the given HSL values.
+    /// 
+    /// **NOTE:** The values given are similar to those used by [`sass`] but are only floating point values. 
+    /// This means:
+    /// 
+    /// * 0.0 <= hue <= 360.0
+    /// * 0.0 <= saturation <= 1.0
+    /// * 0.0 <= lightness <= 1.0
+    /// 
+    /// ```rust
+    /// use turtle::Color;
+    /// let color_rgb = Color::rgb(148.0, 139.0, 193.0);
+    /// let color_hsl = Color::hsl(250.0, 0.3, 0.65);
+    /// assert_eq!(color_rgb, color_hsl);
+    /// ```
+    /// [`sass`]: http://sass-lang.com/documentation/Sass/Script/Functions.html#hsl-instance_method
+    pub fn hsl(hue: f64, saturation: f64, lightness: f64) -> Self {
+        Color::hsla(hue, saturation, lightness, 1.0)
+    }
+
+    /// Create a new `Color` from the given HSL values and the given alpha value.
+    /// 
+    /// **NOTE:** The values given are similar to those used by [`sass`] but are only floating point values. 
+    /// This means:
+    /// 
+    /// * 0.0 <= hue <= 360.0
+    /// * 0.0 <= saturation <= 1.0
+    /// * 0.0 <= lightness <= 1.0
+    /// * 0.0 <= alpha <= 1.0
+    /// 
+    /// ```rust
+    /// use turtle::Color;
+    /// let color_rgba = Color::rgba(148.0, 139.0, 193.0, 1.0);
+    /// let color_hsla = Color::hsla(250.0, 0.3, 0.65, 1.0);
+    /// assert_eq!(color_rgba, color_hsla);
+    /// ```
+    /// [`sass`]: http://sass-lang.com/documentation/Sass/Script/Functions.html#hsla-instance_method
+    pub fn hsla(hue: f64, saturation: f64, lightness: f64, alpha: f64) -> Self {
+        // Most of this code comes courtesy of work done by 'killercup' on GitHub (MIT Licensed) and
+        // the answer here: https://stackoverflow.com/a/9493060.
+        // Link: http://killercup.github.io/hsl-rs/src/hsl/lib.rs.html#111
+        if saturation == 0.0 {
+            let achromatic = (lightness * 255.).round();
+            return Color::rgba(achromatic, achromatic, achromatic, alpha)
+        }
+
+        let hue_to_rgb = |p, q, t| {
+            let t = if t < 0. {
+                t + 1.
+            } else if t > 1. {
+                t - 1.
+            } else { t };
+
+            if t < 1./6. {
+                p + (q - p) * 6. * t
+            } else if t < 1./2. {
+                q
+            } else if t < 2./3. {
+                p + (q - p) * (2./3. - t) * 6.
+            } else {
+                p
+            }
+        };
```

Comment:
Since this function isn't really capturing any variables, let's just define it as an inner function instead of a closure. Also, what do you think about writing it using match as shown below? I think it's quite a bit easier to read this way.

```rust
fn hue_to_rgb(p: f64, q: f64, t: f64) -> f64 {
    // Normalize t
    let t = match t {
        t if t < 0.0 => t + 1.0,
        t if t > 1.0 => t - 1.0,
        t => t,
    };

    match t {
        t if t < 1.0/6.0 => p + (q - p) * 6.0 * t,
        t if t < 1.0/2.0 => q,
        t if t < 2.0/3.0 => p + (q - p) * (2.0/3.0 - t) * 6.0,
        _ => p,
    }
}
```

`fn` declarations are fine within other functions, so you can just replace this variable with the function above.